utils/appliance/implementations/ssui.py

```diff
@@ -20,22 +23,47 @@ def __init__(self, selenium, endpoint, extra_objects=None):
         })
         super(MiqSSUIBrowser, self).__init__(
             selenium,
-            plugin_class=DefaultPlugin,
+            plugin_class=MiqSSUIBrowserPlugin,
             logger=create_sublogger('MiqSSUIBrowser'),
             extra_objects=extra_objects)
+        self.window_handle = selenium.current_window_handle
 
     @property
     def appliance(self):
         return self.extra_objects['appliance']
 
     def create_view(self, *args, **kwargs):
-        return self.appliance.browser.create_view(*args, **kwargs)
+        return self.appliance.ssui.create_view(*args, **kwargs)
 
     @property
     def product_version(self):
         return self.appliance.version
 
 
+class MiqSSUIBrowserPlugin(DefaultPlugin):
+    ENSURE_PAGE_SAFE = (
+        '''if(typeof $ == 'undefined') { return true; } else { return !($.active > 0); }''')
```

Comment:
@mfalesni : Something more needs to be added here I think . the script returns True and allows to login but soon after login the screen exits . It doesn't stay at the login page .

Reply:
So, it seems that after login, sleep will be needed because I did not find anything that would properly detect it so far, but I found that we should probably also check for the progress bar, which should be done something like this, put together with the jquery check:
```javascript
function checkProgressBar() {
    try {
        return $('#ngProgress').attr('style').indexOf('width: 0%') > -1;
    } catch(err) {
        // Not ready yet
        return false;
    }
}

function checkJquery() {
    if(typeof $ == 'undefined') {
        return true;
    } else {
        return !($.active > 0);
    }
}

return checkProgressBar() && checkJquery();
```

Please, do:
```python
from jsmin import jsmin
```

and then use `jsmin` on the script: `ENSURE_PAGE_SAFE = jsmin(...)`